doc/spec/manifest.md

```diff
@@ -0,0 +1,215 @@
+
+    An array of objects which describe content which this application depends
+    on. Those may be generic blobs of data or other content manifests and
+    their dependencies as described by their media type. Fields are:
+    
+    - **`mediaType`** *string*
+    
+        The MIME type of the dependency object.
+    
+    - **`size`** *int*
+    
+        The size in bytes of the dependency object.
+    
+    - **`repository`** *string*
+    
+        The name of the repository from which to fetch the object.
+    
+    - **`digest`** *string*
+    
+        A string labeled with a digest algorithm prefix and the hex-encoded
+        value of that digest for the content. If the content is encrypted,
+        then the digest is of the encrypted content.
+
+        Digest strings should be of the form:
+
+            <digest-alg-name>:<hex-encoded-digest>
+    
+    Any media type may be in the `mediaType` field as the application may
+    use any type of content. If the `mediaType` is
+    `application/vnd.docker.distribution.content.manifest.v2+json` then the
+    dependency is another content manifest as described by this document. The
+    dependencies from that manifest should also be considered dependencies of
+    this application.
+    
+    If the `repository` field is empty or omitted, then the object should be
+    considered to be in the same repository as this manifest.
+
+    In general, the ordering of these dependencies is significant as it may be
+    significant to the application.
+
+- **`application`** *object*
+
+    Fields are:
+    
+    - **`mediaType`** *string*
+    
+        The MIME type of the JSON object in the `data` field.
+    
+    - **`data`** *object*
+    
+        Abstract application data in any schema, as described by the
+        `mediaType` field.
+    
+    Application data may be *any* JSON object, allowing this manifest format,
+    along with the generic dependencies list, to support any type of 
+    application that can be represented as a JSON configuration and a
+    collection of content-addressable blobs of data.
+
+- **`meta`** *object*
+
+    Meta may be *any* JSON object, allowing content creators to annotate
+    their application manifest with any additional labels beyond those already
+    defined by this manifest format. 
+
+### Example Manifest
+
+```json
+{
+    "schemaVersion": 2,
+    "createdAt": "2015-02-26T12:46:31.965839329-08:00",
+    "maintainers": [
+        {
+            "name": "Phillip Edward Nuñez",
+            "email": "pnunez@example.com",
+            "url": "https://example.com/pnunez/"
+        }
+    ],
+    "description": "Application to Simulate Turbo Encabulators",
+    "keywords": [
+        "foo",
+        "bar",
+        "baz"
+    ],
+    "website": "https://foo.bar.baz.example.com",
+    "licenses": [
+        "Apache-2.0"
+    ],
+    "dependencies": [
+        {
+            "mediaType": "application/vnd.docker.distribution.content.manifest.v2+json",
```

Comment:
from above:

> If the `repository` field is empty or omitted, then the object should be considered to be in the same repository as this manifest.

So if I got this manifest as `docker.com/nmcc/foo@sha256:8ebeb6276a15afe71d3c48f0d1801638f0ef2b6db4111a49cc93525093b17f4d` then any dependencies listed here which do not specify a `repository` should be fetched from the `docker.com/nmcc/foo` repository as well.

This dependency has `repository` specified though, so it would be fetched as `isv.example.com/base/system@sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b86451d9bce1f432a537bc7`